server/botany/views.py

```diff
@@ -129,31 +153,7 @@ def play(request, bot1_id, bot2_id):
         state = None
 
         if not game_mod.check_winner(board):
-            bot_mod = loader.create_module_from_str("bot", bot.code)
-            fn = bot_mod.get_next_move
-            param_list = runner.get_param_list(fn)
-            if "state" in request.POST:
-                state = json.loads(b64decode(request.POST["state"]))
-
-            all_args = {
-                "board": board,
-                "move_list": move_list,
-                "token": other_token,
-                "state": state,
-            }
-
-            args = {
-                param: value for param, value in all_args.items() if param in param_list
-            }
-
-            rv = fn(**args)
-            try:
-                next_next_move, state = rv
-                print("Got a state")
-            except TypeError:
-                next_next_move, state = rv, None
-                print("Got no state")
-            move_list.append(next_next_move)
+            board, move_list = _make_bot_move(game_mod,bot,board,move_list,other_token,state)
```

Comment:
You completely dropped state handling here. The original code would parse out the state with `json.loads(b64decode(request.POST["state"]))`. This breaks bots that keep a state.